src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/CompletionListCache.cs

```diff
@@ -3,68 +3,71 @@
 
 using System;
 using System.Collections.Generic;
-using Microsoft.CodeAnalysis.Razor.Completion;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.Completion
 {
     internal sealed class CompletionListCache
     {
         // Internal for testing
-        internal static readonly int MaxCacheSize = 3;
+        internal static readonly int MaxCacheSize = 10;
 
         private readonly object _accessLock;
-        private readonly List<(long, IReadOnlyList<RazorCompletionItem>)> _resultIdToCompletionList;
-        private long _nextResultId;
+        private readonly List<CacheEntry> _resultIdToCacheEntry;
+        private int _nextResultId;
```

Comment:
No where else in our system utilized long. `int` is definitely enough and if it overflows (somehow) that's totally fine too.